hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/strategies/VersionedEngagementsStrategy.java

```diff
@@ -75,13 +77,8 @@ public List<Finding> run(Scan scan, ScanFile scanResultFile) throws Exception {
       LOG.debug("Using configured User Id");
       userId = this.config.getUserId();
     } else {
-      if (this.persistenceProviderConfig.isInLowPrivilegedMode()) {
-        throw new DefectDojoPersistenceException("You need to configure the id of the DefectDojo user when running the DefectDojo Hook in low privileged mode. See: https://www.securecodebox.io/docs/hooks/defectdojo#low-privileged-mode");
-      }
-      LOG.debug("Getting DefectDojo User Id via Username from the API");
-      userId = userService.searchUnique(User.builder().username(this.config.getUsername()).build())
-        .orElseThrow(() -> new DefectDojoPersistenceException("Failed to find user with name: '" + this.config.getUsername() + "'"))
-        .getId();
+      LOG.debug("Getting DefectDojo User Id via user profile API");
+      userId = userProfileService.search().get(0).getUser().getId();
```

Comment:
This looks like a possible IOOBE: If the collection returned by `search()` the `get(0)` will throw an `IndexOutOfBoundException.